.github/workflows/main.yml

```diff
@@ -20,74 +20,14 @@ jobs:
     runs-on: ubuntu-latest
 
     steps:
-    - name: Memory report
-      run: |
-        echo "Memory and swap:"
-        free
-        echo
-        swapon --show
-        echo
-        echo "Available storage:"
-        df -h
-        echo
-    - name: Add 8G more swap for native-image build
-      run: |
-        sudo dd if=/dev/zero of=/mnt/swapfile-2 bs=1MiB count=$((8*1024))
-        sudo mkswap /mnt/swapfile-2
-        sudo swapon /mnt/swapfile-2
     - uses: actions/checkout@v2
-    - name: Set up JDK 11
-      uses: actions/setup-java@v1
+    - name: Setup runner
+      uses: ./.github/actions/setup-runner
+    - name: Setup Java, Maven, Gradle
+      uses: ./.github/actions/dev-tool-java
```

Comment:
how are things ordered in the composite actions? Do we have any ordering requirements?

Reply:
Not really. It's more a "it feels more natural to setup the runner first" thing.

Reply:
ah, no sorry. I meant in the dev-tool-java action. That is a composite action, is the order of runs defined?

Reply:
It's all sequential.